File: Inheritance/Person.m
//
//  Person.m
//  Inheritance
//
//  Created by Mark Wong on 2022-05-12.
//

#import "Person.h"

@implementation Person

- (instancetype)initWithName: (NSString *) name andAge: (NSInteger) age
{
  self = [super init];
  if (self) {
    _name = name;
    _age = &age;
  }
  return self;
}

- (void) printInfo
{
  NSLog(@"Name: %@, Age: %d", _name, _age);
}

@end

File: ClassAndProperties/Person.swift
//
//  Person.swift
//  ClassAndProperties
//
//  Created by Mark Wong on 2022-05-10.
//

import Foundation

class Person {
  // property(instance variable + getter/setter)
  private(set) var name: String
  private(set) var age: Int
  
  init(name: String, age: Int) {
    self.name = name
    self.age = age
  }
  
  func printPersonInfo() {
    print("Name: \(name), Age: \(age)")
  }
}
